nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -657,6 +653,25 @@ in {
   };
 
   config = mkIf cfg.enable {
+    assertions = [
+      { assertion = hasLocalPostgresDB -> config.services.postgresql.enable;
+        message = ''
+          Cannot deploy matrix-synapse with a configuration for a local postgresql database
+            and a missing postgresql service. Since 20.03 it's mandatory to manually configure the
```

Comment:
Indentation gone wrong here?

Reply:
I did this on purpose: the first line will be prefixed with a `- `, so with my indention the error message will actually look like this:

```
error:
Failed assertions:
- Cannot deploy matrix-synapse with a configuration for a local postgresql database
  and a missing postgresql service. Since 20.03 it's mandatory to manually configure the
  database (please read the thread in https://github.com/NixOS/nixpkgs/pull/80447 for
  further reference).

  If you
  - try to deploy a fresh synapse, you need to configure the database yourself. An example
    for this can be found in <nixpkgs/nixos/tests/matrix-synapse.nix>
  - update your existing matrix-synapse instance, you simply need to add `services.postgresql.enable = true`
    to your configuration.

For further information about this update, please read the release-notes of 20.03 carefully.

(use '--show-trace' to show detailed location information)
```